toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -499,24 +499,26 @@ func calculateTotalPackages(packages []string, installRoot string) (totalPackage
 	return
 }
 
+// addMachineID creates the /etc/machine-id file in the installChroot
 func addMachineID(installChroot *safechroot.Chroot) (err error) {
+	// From https://www.freedesktop.org/software/systemd/man/machine-id.html:
+	// For operating system images which are created once and used on multiple
+	// machines, for example for containers or in the cloud, /etc/machine-id
+	// should be an empty file in the generic file system image. An ID will be
+	// generated during boot and saved to this file if possible.
+
 	const (
-		squashErrors = false
-		setupProgram = "/bin/systemd-machine-id-setup"
+		squashErrors       = false
+		setupProgram       = "install"
+		machineIDFile      = "/etc/machine-id"
+		machineIDFilePerms = "0644"
 	)
 
-	// Check if systemd-machine-id-setup is present before invoking it,
-	// some images will not use systemd (such as a container)
-	exists, _ := file.PathExists(filepath.Join(installChroot.RootDir(), setupProgram))
-	if !exists {
-		logger.Log.Debugf("'%s' not found inside chroot '%s', skipping adding machine ID", setupProgram, installChroot.RootDir())
-		return
-	}
-
 	ReportAction("Configuring machine id")
 
 	err = installChroot.UnsafeRun(func() error {
-		return shell.ExecuteLive(squashErrors, setupProgram)
+		setupArgs := []string{"-m", machineIDFilePerms, "/dev/null", machineIDFile}
+		return shell.ExecuteLive(squashErrors, setupProgram, setupArgs...)
```

Comment:
Do you need to shell out to create an empty file with a given permission?  Can you use `f, err := os.OpenFile("/etc/machine-id", os.O_CREATE, 0644); f.Close()` instead?

Reply:
This shell method was done to minimize the PR deltas, but since we are just creating a blank file now, it does seem to be cleaner overall to just use Go's file creation APIs and remove the dependence on shell/coreutils. Will update.

Reply:
Updated.